custom-attachments/src/main/java/io/getstream/chat/android/customattachments/activity/MessagesActivity.kt

```diff
@@ -0,0 +1,66 @@
+package io.getstream.chat.android.customattachments.activity
+
+import android.content.Context
+import android.content.Intent
+import com.google.android.material.datepicker.MaterialDatePicker
+import io.getstream.chat.android.client.models.Attachment
+import io.getstream.chat.android.customattachments.R
+import io.getstream.chat.android.customattachments.factory.DateAttachmentPreviewFactory
+import io.getstream.chat.android.ui.message.MessageListActivity
+import io.getstream.chat.android.ui.message.MessageListFragment
+import io.getstream.chat.android.ui.message.input.MessageInputView
+import java.text.DateFormat
+import java.util.Date
+
+class MessagesActivity : MessageListActivity() {
+
+    override fun createMessageListFragment(cid: String, messageId: String?): MessageListFragment {
+        return MessageListFragment.newInstance(cid) {
+            setFragment(CustomMessageListFragment())
+            customTheme(R.style.StreamUiTheme)
+            showHeader(true)
+            messageId(messageId)
+        }
+    }
+
+    class CustomMessageListFragment : MessageListFragment() {
+
+        override fun setupMessageInput(messageInputView: MessageInputView) {
+            super.setupMessageInput(messageInputView)
+
+            messageInputView.setAttachmentButtonClickListener {
```

Comment:
Do you think it might be good for us to leave step-by-step comments here to break the monotony into smaller, clear steps?

It could be something like:

// 1. - Build a date picker
// 2. - Add a listener that will add a date attachment on positive click
// 3. - Show the  dialog

I know that what you are doing here is pretty standard but

a) We can't guarantee that the person reading this knows about it
b) Larger chunks of code tend to confuse people

Reply:
Btw. I am only saying this about the repo itself, I understand the guides are written in a different manner.

These comments would be intended for people who like to browse repositories.

Reply:
Agreed - adding smaller bits of comments would help people who might've skipped the docs part know what's going on